presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -82,7 +84,7 @@
     private final Map<NodeRef<Expression>, FieldId> columnReferences = new LinkedHashMap<>();
 
     // a map of users to the columns per table that they access
-    private final Map<Identity, Map<QualifiedObjectName, Set<String>>> tableColumnReferences = new LinkedHashMap<>();
+    private final Map<AccessControlInfo, Map<QualifiedObjectName, Set<String>>> tableColumnReferences = new LinkedHashMap<>();
```

Comment:
ok.   I'll create an enum to represent access type.  Values are `ALLOW/VIEW/DEFAULT`.   

`select` and `delete` would be misleading names because not all select queries would use the `select` access control (views).  And lots of other privileges would also use it (including delete for some things).

Reply:
I'm not sure this makes sense from an analysis standpoint. We want to report facts about the type of column reference, not how they should be access checked later. The type of access can be translated into an access check policy, but shouldn't be specific to access control.